### PR TITLE
Fixing OAuth / OIDC support

### LINKED ISSUE #1945: OIDC support: discovery endpoint is producing wrong field names
When using the OpenID Connect server support, the discovery url does not return the `subject_types_supported` field.
In the spec, it says it SHOULD and not MUST so it doesn't seem to be mandatory, though there is an ambiguity which leads implementations to require it or not.

In the pac4j `OidcClient` (based on the Nimbus SDK), it fails due to this missing field.

After some discussion with his creator, the recommendation would be to add this field, while the spec is fixed.

### PR DESCRIPTION
**Pull request by leleuj**
Closes #1945

### CODE REVIEW

File: cas-server-support-oauth/src/main/java/org/apereo/cas/config/CasOAuthConfiguration.java [hunk 1]
```diff
@@ -395,6 +396,7 @@ public void initializeServletApplicationContext() {
             service.setDescription("OAuth Wrapper Callback Url");
             service.setServiceId(oAuthCallbackUrl);
             service.setEvaluationOrder(Integer.MIN_VALUE);
+            service.setAttributeReleasePolicy(new ReturnAllAttributeReleasePolicy());
```
Comment (leleuj) [inline cas-server-support-oauth/src/main/java/org/apereo/cas/config/CasOAuthConfiguration.java]: The callback url service should return ALL services as the filtering is done here: https://github.com/apereo/cas/blob/master/cas-server-support-oauth/src/main/java/org/apereo/cas/support/oauth/web/BaseOAuthWrapperController.java#L101 for the real OAuth / OIDC service.

File: cas-server-support-oidc/src/main/java/org/apereo/cas/config/OidcConfiguration.java [hunk 2]
```diff
@@ -106,6 +106,18 @@
     @Qualifier("defaultOAuthCodeFactory")
     private OAuthCodeFactory defaultOAuthCodeFactory;
 
+    @Autowired
+    @Qualifier("oidcAuthorizeEndpointController")
```
Comment (mmoayyed) [inline cas-server-support-oidc/src/main/java/org/apereo/cas/config/OidcConfiguration.java]: This is an anti-pattern. All these should be removed and moved inside the bean itself. To fix the Controller annotation issue, try removing the BaseOAuthWrapperController as a return type and return the actual bean type.

**Comment by mmoayyed** [comment]
Great. Thanks very much!

File: cas-server-documentation/installation/OIDC-Authentication.md [hunk 3]
```diff
@@ -49,6 +49,7 @@ OpenID Connect clients can be registered with CAS as such:
 | `serviceId`                   | The redirect URI for this OIDC client.
 | `signIdToken`                 | Whether ID tokens should be signed.
 | `jwks`                        | Path to the location of the keystore that holds the signing keys. If none defined, default below will be used.
+| `idTokenSigningAlgValues`     | The supported algorithms for the signature of the ID Token. If none defined, RS256 will be used.
```
Comment (mmoayyed) [inline cas-server-documentation/installation/OIDC-Authentication.md]: Not a big deal, but a shorter name perhaps: `idTokenSigningAlgs`.

File: cas-server-support-oidc/src/main/java/org/apereo/cas/config/OidcConfiguration.java (hunk 2 above)
Comment (leleuj) [inline cas-server-support-oidc/src/main/java/org/apereo/cas/config/OidcConfiguration.java]: OK

File: cas-server-documentation/installation/OIDC-Authentication.md (hunk 3 above)
Comment (leleuj) [inline cas-server-documentation/installation/OIDC-Authentication.md]: OK. Will do.

**Comment by leleuj** [comment]
Accounted for @mmoayyed 's comments

**Comment by mmoayyed** [comment]
Fantastic. Thanks again.

**Comment by mmoayyed** [comment]
@leleuj Do these endpoints need to be updates https://apereo.github.io/cas/development/installation/OIDC-Authentication.html ?

**Comment by mmoayyed** [comment]
Also, I forget: did you update the mgmt app to account for the new OIDC field you added?

**Comment by leleuj** [comment]
Yes, I forgot both. I'll submit a new PR for that.

**Pull request merged**